keras_cv/visualization/plot_image_gallery.py

```diff
@@ -92,27 +89,87 @@ def unpackage_tfds_inputs(inputs):
             "to be true."
         )
 
-    fig = plt.figure(figsize=(cols * scale, rows * scale))
-    fig.tight_layout()  # Or equivalently,  "plt.tight_layout()"
-    plt.subplots_adjust(wspace=0, hspace=0)
-    plt.margins(x=0, y=0)
-    plt.axis("off")
+    def unpack_images(inputs):
+        return inputs["image"]
+
+    # Calculate appropriate number of rows and columns
+    if rows is None and cols is None:
+        if isinstance(images, tf.data.Dataset):
+            sample = next(iter(images.take(1)))
+            sample_shape = sample["image"].shape
+            if len(sample_shape) == 4:
+                batch_size = sample_shape[0]
+            else:
+                raise ValueError(
+                    "Passed `tf.data.Dataset` does not appear to be batched. Please batch using the `.batch().`"
+                )
+
+            images = images.map(unpack_images)
+            images = images.take(batch_size)
+            images = next(iter(images))
+        else:
+            sample_shape = images.shape
+            if len(sample_shape) == 4:
+                batch_size = sample_shape[0]
+            else:
+                raise ValueError(
+                    f"Passed '`{type(images)}`' does not appear to be batched. Please batch using the `.batch()."
```

Comment:
Let's say

"plot_image_gallery received unbatched images and `cols` and `rows` were both `None`. Either images should be batched, or `cols` and `rows` should be specified.

Reply:
Done!